test/unit.js

```diff
@@ -0,0 +1,273 @@
+            });
+
+            it('the react component renders props through the attribute', function () {
+                const WEB_COMPONENT_TEXT = 'terminal component';
+
+                renderApp(
+                    React.createClass({
+                        render: function () {
+                            return <ReactComponent text={WEB_COMPONENT_TEXT}/>;
+                        }
+                    })
+                );
+
+                expect(getWebComponent().firstChild.textContent).to.equal(WEB_COMPONENT_TEXT);
+            });
+
+            it('changing the state changes the web component attribute', function () {
+                var renderedApp = renderApp(
+                    React.createClass({
+                        getInitialState: function () {
+                            return {
+                                text: 'initial text'
+                            }
+                        },
+                        render: function () {
+                            return <ReactComponent text={this.state.text}/>;
+                        }
+                    })
+                );
+
+                expect(getWebComponent().firstChild.textContent).to.equal('initial text');
+                renderedApp.setState({
+                    text: 'updated text'
+                });
+                expect(getWebComponent().firstChild.textContent).to.equal('updated text');
+            });
+
+            it('the react component\'s display is the camel cased version of the tag name', function () {
+                expect(camelCase(tagName)).to.equal(ReactComponent.displayName);
+            });
+        });
+
+        describe('that has a property but not an attribute;', function () {
+            var ReactComponent;
+
+            beforeEach(function () {
+                tagName = helperElement('x-terminal-hidden').safe;
+                var webComponentConstructor = skate(tagName, {
+                    render: function (elem) {
+                        elem.innerHTML = '<div></div>';
+                        if (elem.hidden) {
+                            elem.firstChild.setAttribute('aria-hidden', '');
+                        } else {
+                            elem.firstChild.removeAttribute('aria-hidden');
+                        }
+                    },
+                    properties: {
+                        hidden: {
+                            set: function (elem) {
+                                skate.render(elem);
+                            }
+                        }
+                    }
+                });
+
+                ReactComponent = makeReactComponent(webComponentConstructor)
+            });
+
+            it('the web component renders props through the property', function () {
+                renderApp(
+                    React.createClass({
+                        render: function () {
+                            return <ReactComponent hidden="true"/>;
```

Comment:
Not sure if you meant this to be a boolean true, but you can do boolean attributes in jsx by: `hidden={true}`.